### PR TITLE
#2719 converted Matcher.java to Kotlin

### LINKED ISSUE #2719: convert matcher.java to Kotlin
Parent issue: #1826

What changes you have made:
Converted Java file to Kotlin

### PR DESCRIPTION
**Pull request by cdhiraj40**
Fixes #2719

### CODE REVIEW

**Comment by kelson42** [comment]
@gouri-panda any feedback/review about this PR?

**Comment by gouri-panda** [comment]
@kelson42 Sorry I was busy with other things. I will do this.

File: app/src/androidTest/java/org/kiwix/kiwixmobile/testutils/Matcher.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+package org.kiwix.kiwixmobile.testutils
+
+import android.view.View
+import android.view.ViewGroup
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.TypeSafeMatcher
+
+object Matcher {
+  fun childAtPosition(
```
Comment (gouri-panda) [inline app/src/androidTest/java/org/kiwix/kiwixmobile/testutils/Matcher.kt:28]: Should be a static method.

**Comment by kelson42** [comment]
@cdhiraj40 Any feedback?

File: app/src/androidTest/java/org/kiwix/kiwixmobile/testutils/Matcher.kt (hunk 1 above)
Comment (cdhiraj40) [reply app/src/androidTest/java/org/kiwix/kiwixmobile/testutils/Matcher.kt:28]: okay, any preferable class name?

**Comment by cdhiraj40** [comment]
> @cdhiraj40 Any feedback?

sorry for delay

**Comment by cdhiraj40** [comment]
i have done the changes do check :)

File: app/src/androidTest/java/org/kiwix/kiwixmobile/language/LanguageFragmentTest.java [hunk 2]
```diff
@@ -97,8 +97,8 @@ public void testLanguageFragment() {
     // Get a reference to the checkbox associated with the top selected language
     checkBox1 = onView(
       allOf(withId(R.id.item_language_checkbox),
-        childAtPosition(
-          childAtPosition(
+        Matcher.Companion.childAtPosition(
+          Matcher.Companion.childAtPosition(
```
Comment (gouri-panda) [inline app/src/androidTest/java/org/kiwix/kiwixmobile/language/LanguageFragmentTest.java:101]: Use Kotlin's Jvm static annotation in the method of the class.

File: app/src/androidTest/java/org/kiwix/kiwixmobile/language/LanguageFragmentTest.java [hunk 3]
```diff
@@ -43,7 +43,6 @@
 import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
 import static androidx.test.espresso.matcher.ViewMatchers.withContentDescription;
 import static androidx.test.espresso.matcher.ViewMatchers.withId;
-import static androidx.test.espresso.matcher.ViewMatchers.withText;
```
Comment (gouri-panda) [inline app/src/androidTest/java/org/kiwix/kiwixmobile/language/LanguageFragmentTest.java:46]: Why is it necessary here?

**Comment by gouri-panda** [comment]
@cdhiraj40 Please rebase all  the commits.

File: app/src/androidTest/java/org/kiwix/kiwixmobile/language/LanguageFragmentTest.java (hunk 3 above)
Comment (cdhiraj40) [reply app/src/androidTest/java/org/kiwix/kiwixmobile/language/LanguageFragmentTest.java:46]: the import was unused, should i add it again?

Comment (gouri-panda) [reply app/src/androidTest/java/org/kiwix/kiwixmobile/language/LanguageFragmentTest.java:46]: Yes! It's not related to PR.

**Comment by cdhiraj40** [comment]
> @cdhiraj40 Please rebase all the commits.

I think I did, can you check?

**Review by gouri-panda — APPROVED** [review]
@cdhiraj40 The file isn't properly formatted, but if you have the time you can do this :)

**Review by gouri-panda — APPROVED** [review]

**Comment by gouri-panda** [comment]
@kelson42 The PR is ready to merge :)

**Pull request merged**